dialer.go

```diff
@@ -456,7 +457,10 @@ func (d *Dialer) Warmup(ctx context.Context, icn string, opts ...DialOption) err
 	for _, opt := range opts {
 		opt(&cfg)
 	}
-	_, err = d.connectionInfoCache(ctx, cn, &cfg.useIAMAuthN)
+	c, err := d.connectionInfoCache(ctx, cn, &cfg.useIAMAuthN)
```

Comment:
The error return from connection info cache is really only interesting if the key generator fails. If an IAM permission is missing, the call to `d.connectionInfoCache` won't return an error until someone asks for `ConnectionInfo`. So as written there's no way to properly close the internal cache.

What if we just replaced these internals with a call to `EngineVersion`? That would solve all the problems above and make use of the error checking you've added.

Reply:
I see the API signatures don't support this, so we'll have to just match the internal implementation between EngineVersion and Warmup.

Reply:
As discussed offline, I have made `Warmup` call `ConnectionInfo` as there is no other way to invalidate without breaking the API.